src/server/main_service.cc

```diff
@@ -1903,8 +1903,10 @@ void Service::Exec(CmdArgList args, ConnectionContext* cntx) {
     }
   }
 
-  if (exec_info.preborrowed_interpreter)
+  if (exec_info.preborrowed_interpreter) {
```

Comment:
how did you reproduce this? can you create a pytest?

Reply:
I am not sure it will be easy to make it run in CI due to timeouts of the tests, running on 2 cores machine. I was debugging this a lot.
From what I understand the crash is when running different lua scripts inside multi, is this correnct?
you might be able to reproduce this in unit test as well

Reply:
Sure, you're right. I did even better - repro in a unittest and made sure it fails before and is fixed after.